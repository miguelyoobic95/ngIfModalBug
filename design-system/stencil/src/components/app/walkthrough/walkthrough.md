---
name: Walkthrough
category: App Components
---

```walkthrough.html
    <yoo-walkthrough></yoo-walkthrough>
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`entry`|`Array<{title: string, img: string}>`|   |entry of the component|

## CSS

|Type|Name|Description|
|---|---|---|


## Events
|Attr|Description|
|---|---|---|