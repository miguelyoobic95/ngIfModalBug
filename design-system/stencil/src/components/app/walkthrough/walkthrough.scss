:host {
    background-color: cssvar(light);
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    ion-slides {
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
    }
    .slide-zoom {
        height: 100%;
    }
    .swiper-slide {
        display: block;
    }
    /deep/ .swiper-container {
        height: 100%;
    }
    .slide-content {
        padding: $padding;
        height: 100%;
        img {
            max-height: 200px;
            margin-bottom: $padding;
        }
        .slide-title {
            margin-top: 2.8rem;
            font-size: 1.25rem;
            font-weight: 600;
            margin-bottom: 1rem;
            color: cssvar(black);
        }
        .slide-subtitle {
            color: cssvar(text-color);
        }
    }
    /deep/ .swiper-pagination-bullet-active {
        background-color: cssvar(success) !important;
    }
}