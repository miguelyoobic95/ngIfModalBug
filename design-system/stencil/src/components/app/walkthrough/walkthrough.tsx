import { Component, Element, Prop, Method, Event, EventEmitter } from '@stencil/core';
import { IWalkthroughEntry, IWalkthroughSlideEvent } from '@shared/interfaces';

@Component({
    tag: 'yoo-walkthrough',
    styleUrl: 'walkthrough.scss',
    scoped: true
})
export class YooWalkthroughComponent {

    @Prop() config: Array<IWalkthroughEntry>;

     @Element() host: HTMLStencilElement;

    @Event() slideChanged: EventEmitter<IWalkthroughSlideEvent>;

    private ionSlides: HTMLIonSlidesElement;

    componentDidLoad() {
        this.ionSlides = this.host.querySelector('ion-slides');
    }

    @Method()
    update() {
        if (this.ionSlides) {
            this.ionSlides.update();
        }
    }

    @Method()
    slideNext() {
        if (this.ionSlides) {
            this.ionSlides.slideNext();
        }
    }

    @Method()
    isEnd() {
        if (this.ionSlides) {
            return this.ionSlides.isEnd();
        }
        return false;
    }

    @Method()
    lockSwipes(shouldLock: boolean) {
        if (this.ionSlides) {
            this.ionSlides.lockSwipes(shouldLock);
        }
    }

    onIonSlideDidChange(ev: CustomEvent) {
        let index = ev.detail.activeIndex;
        let slide = this.config[index];
        this.slideChanged.emit({ event: ev.detail, slide });
    }

    render(): JSX.Element {
        let config = this.config || [];
        return (
            <ion-slides onIonSlideDidChange={ev => this.onIonSlideDidChange(ev)}>
                {
                    config.map(s =>
                        <ion-slide>
                            <div class="slide-content" attr-layout="column" attr-layout-align="center center">
                                <div>
                                    <div class="image-container">
                                        <img src={s.imageUrl} />
                                    </div>
                                    <div class="text-container">
                                        <div class="slide-title" innerHTML={s.title}></div>
                                        <div class="slide-subtitle" innerHTML={s.subtitle}></div>
                                    </div>
                                </div>
                            </div>
                        </ion-slide>
                    )
                }
            </ion-slides>
        );
    }
}
