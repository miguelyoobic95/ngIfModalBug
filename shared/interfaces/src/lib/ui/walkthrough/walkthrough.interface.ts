export interface IWalkthroughEntry {
    title: string;
    subtitle?: string;
    imageUrl: string;
    type?: string;
}

export interface IWalkthroughSlideEvent {
    event: any;
    slide: IWalkthroughEntry;
}