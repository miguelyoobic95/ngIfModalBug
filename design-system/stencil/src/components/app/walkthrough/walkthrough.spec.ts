import { TestWindow } from '@stencil/core/testing';
import { YooWalkthroughComponent } from './walkthrough';

describe('WalkthroughComponent', () => {
    it('should build', () => {
        expect(new YooWalkthroughComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach( async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooWalkthroughComponent],
                html: '<yoo-walkthrough></yoo-walkthrough>'
            });
        });

        it ('Should render', async () => {
            expect(element).toMatchSnapshot();
        });
    });

    describe('Events', () => {

        let emitEvent, window;
        beforeEach( async () => {
            window = new TestWindow();
            emitEvent = jest.fn();
        });
    });
});