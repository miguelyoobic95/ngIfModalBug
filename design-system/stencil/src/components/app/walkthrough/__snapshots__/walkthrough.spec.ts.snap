// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalkthroughComponent Rendering Should render 1`] = `
<yoo-walkthrough
  class="hydrated"
  data-yoo-walkthrough-host=""
>
  
  <ion-slides
    data-yoo-walkthrough=""
  />
</yoo-walkthrough>
`;
